lib/match/array_match.rb

```diff
@@ -0,0 +1,14 @@
+module Match
+  module ArrayMatch
```

Comment:
Dry up the naming.  `RuboClaus::Match::Array or just `RuboClaus::Array` with a `#match` method.  Again, just private methods in an included module is odd.  Dont move stuff around just to get it out of a file.  Move stuff around if it makes more sense to be logically structured in that location and/or is more composable and reusable.